### PR TITLE
ACP-3963: Algolia personalization.

### PR DESCRIPTION
**Pull request by matweew**
## PR Description
Ticket: https://spryker.atlassian.net/browse/ACP-4336

Algolia events and personalization Integration Guides when you have time, it also includes updates for the main Algolia page, and Configure page and has an updated search integration guide.

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-headless.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  * https://www.algolia.com/doc/guides/building-search-ui/getting-started/angular/
+* Flutter:
+  * https://www.algolia.com/doc/guides/building-search-ui/getting-started/flutter/
+
+For mobile applications:
+* Android https://www.algolia.com/doc/guides/building-search-ui/getting-started/android/
+* iOS https://www.algolia.com/doc/guides/building-search-ui/getting-started/ios/
+
+
+For additional "conversion" events such as "Add to Cart", "Add to Shopping List", or "Add to Wishlist" use Insights API client for your framework/platform
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-headless.md:45]:
```suggestion
For additional "conversion" events such as "Add to Cart", "Add to Shopping List", or "Add to Wishlist" use the Insights API client for your framework/platform
```

Are those conversion events specific to Spryker? Will there be more docs about those? When to use, where, etc?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+If you plan to use Algolia personalization in headless frontend or mobile application follow [this guide](/docs/pbc/all/search/{{page.version}}/base-shop/third-party-integrations/algolia/algolia-personalization-headless.html). 
+
+This feature also enables other Algolia premium features:
+
+- Personalization
+- Dynamic Re-Ranking
+- Query Categorization
+- Search analytics
+- Revenue analytics
+- A/B Testing
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:20]: `This feature` means `Algolia Personalization` I assume `Personalization` can be removed from this list.

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    protected function getEventDispatcherPlugins(): array
+    {
+        return [
+            //...
+            new Spryker\Yves\Customer\Plugin\EventDispatcher\AnonymousIdSessionAssignEventDispatcherPlugin(),
+        ];
+}
+```
+
+4. Build a new javascript assets for Yves `npm ci && npm run yves` or `console frontend:project:install-dependencies && console frontend:yves:build`.
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:91]:
```suggestion
4. Build the JavaScript assets for Yves `npm ci && npm run yves` or `console frontend:project:install-dependencies && console frontend:yves:build`.
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+   Usually it's executed automatically during Spryker Cloud deployment pipeline. But it's better to check this command on local development environment first.
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:92]:
```suggestion
   Usually, it's executed automatically during the Spryker Cloud deployment pipeline. But it's better to check this command on the local development environment first.
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+5. Check your Yves's compatibility with the feature:
+
+In case of customizations your codebase could have updated Yves templates on project level (src/Pyz/Yves/).
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:96]:
```suggestion
In case of customizations, your codebase could have updated Yves templates on the project level (src/Pyz/Yves/).
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+// TODO (@matweew/@supproduction): updated after the implementation of debugger.
+`TraceableEventWidgetConfig::getIsDebugEnabled()` set to `true`.
+  
+* Run the project locally or deploy to testing environment.
+* Open your Shop Storefront home page.
+* Open browser's development console and check "Preserve log" in the settings.
+
+Test the correctness of data in the triggered events in the browser console:
+* Open Home page
+  * (if applicable) Click on a product - `PRODUCT_CLICK`
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:108]: When are those events applicable?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+  * (if applicable) Click on a product add to cart button - `ADD_TO_CART`
+* Open product detail page (PDP), you should see events for actions:
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:110]:
```suggestion
* Open the product detail page (PDP), you should see events for the actions: 
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md [hunk 8]
```diff
@@ -159,11 +165,21 @@ Default attributes for faceting are as follows:
...
 
 The `prices` attribute is an object with nested fields. Algolia creates facets for each nested field and creates facets for all the currencies and pricing modes available in product entities.
 
 #### Facet configuration
 
+##### Facet display
+
+All configured index Facets will be displayed in the list of filter on Yves or via Glue API `/catalog-search`.
+Also, Spryker's Algolia app supports `renderingContent` feature, which can be found in an index "Configuration > Facet display".
+Here you can configure the order of facets and add only relevant for end-users facets.
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md:178]:
```suggestion
Here you can configure the order of facets and add only for end-users relevant facets.
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md [hunk 9]
```diff
@@ -159,11 +165,21 @@ Default attributes for faceting are as follows:
...
+
+Please pay attention that Spryker has additional UI configuration for filters in the Backoffice > Merchandising > Filter Preferences,
+where you can change a filter type: single-select, multi-select or range. That will be applied also for Algolia search results.
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md:181]: What is the impact of each?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md [hunk 10]
```diff
@@ -180,15 +196,21 @@ Setting an attribute as `filter only` prevents it from showing in the facets lis
...
+This checkbox is checked by default. Clearing this checkbox changes the calculation method for facet aggregation for the given field.
+By default, facet aggregation is calculated after search results for a given query are processed by deduplication or grouping process.
+It is not recommended to turn this checkbox off.
 
-#### Algolia and Glue API facets interaction
+#### Spryker Filter Preferences and Algolia app
 
-When used with Algolia PBC, Spryker facets configuration is ignored and Algolia facets configuration is used instead. This means that Glue API response fields `"valueFacets"` and `"rangeFacets"` include facets configured in Algolia. Be wary that setting any Algolia facet to `"filter only"` mode removes it from the corresponding field in Glue API response.
+When used with Algolia app, Spryker facets configuration is ignored and Algolia facets configuration is used instead.
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md:211]:
```suggestion
When used with the Algolia app, the Spryker facets configuration is ignored and the Algolia facets configuration is used instead.
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/integrate-algolia.md [hunk 11]
```diff
@@ -708,21 +704,16 @@ class SynchronizationDependencyProvider extends SprykerSynchronizationDependency
 }
 ```
 
-### Receive ACP messages
```
Comment (stereomon) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/integrate-algolia.md:711]: Is this section moved to another place or is it simply not needed anymore?

**Review by stereomon — CHANGES_REQUESTED** [review]

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-headless.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
+---
+title: Algolia Personalization for headless frontends
+description: Find out how you can enable Algolia personalization in your Spryker shop based on headless approach (custom frontend or mobile application). 
+last_updated: Nov 24, 2024
```
[the remark this answers is not in the document]
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-headless.md:4]: I don't know when we release it, but anyway it will be closer to reality than if I use today :)

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-headless.md (hunk 1 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-headless.md:45]: They are project-specific. Each project decides what conversion events are important for them. It's described in the guide from Algolia

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 6 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:108]: I mean if applicable for the project's home page (if they products are on the home page). Not clear?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md (hunk 9 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md]: only in UI, updated

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/integrate-algolia.md (hunk 11 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/integrate-algolia.md:711]: it's not relevant to this integration guide, it's generic ACP functionality that's provide by default and does not require a developer action.

**Review by stereomon — DISMISSED** [review]

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 13]
```diff
@@ -0,0 +1,328 @@
...
+            //...
+            new Spryker\Yves\Customer\Plugin\EventDispatcher\AnonymousIdSessionAssignEventDispatcherPlugin(),
+        ];
+}
+```
+
+4. Build the JavaScript assets for Yves `npm ci && npm run yves` or `console frontend:project:install-dependencies && console frontend:yves:build`.
+   Usually, it's executed automatically during the Spryker Cloud deployment pipeline. But it's better to check this command on the local development environment first.
+   
+  Check that required modulule`npm list search-insights` if you have `└── (empty)` result it means that you have to install `search-insights` manually `npm i search-insights`.
```
Comment (supproduction) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:96]:
```suggestion
  Check that required dependency is installed `npm list search-insights` if you have `└── (empty)` result it means that you have to install `search-insights` manually `npm i search-insights`.
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia.md [hunk 14]
```diff
@@ -29,6 +29,7 @@ Your users can search for active concrete products by the following attributes:
 - Name
 - Description
 - Keywords
+- or any other attribute that your products have (when it's configured in Algolia Dashboard)
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia.md:32]: Do the previous attributes not need to be configured in Algolia Dashboard?

Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia.md:32]: they're preconfigured by default by ACP Algolia App

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md [hunk 15]
```diff
@@ -159,11 +165,23 @@ Default attributes for faceting are as follows:
...
+- `merchant_name` (for Marketplace shops)
 
 The `prices` attribute is an object with nested fields. Algolia creates facets for each nested field and creates facets for all the currencies and pricing modes available in product entities.
 
 #### Facet configuration
 
+##### Facet display
+
+All configured index Facets will be displayed in the list of filter on Yves or via Glue API `/catalog-search`.
+Also, Spryker's Algolia app supports `renderingContent` feature, which can be found in an index "Configuration > Facet display".
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md:177]: Is `Configuration > Facet display` in Algolia dashboard on back office?

Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/configure-algolia.md:177]: Algolia Dashboard in Search index Configuration

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 16]
```diff
@@ -0,0 +1,360 @@
...
+spryker-shop/checkout-page: ^3.32.1
+spryker-shop/home-page: ^1.2.0
+spryker-shop/payment-page: ^1.5.0
+spryker-shop/product-detail-page: ^3.22.1
+spryker-shop/product-group-widget: ^1.10.1
+spryker-shop/product-review-widget: ^1.16.1
+spryker-shop/product-set-detail-page: ^1.11.0
+spryker-shop/quick-order-page: ^4.10.0
+spryker-shop/shop-ui: ^1.79.0
+spryker-shop/traceable-event-widget: ^1.0.1
```
Comment (matweew) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:65]: @vol4onok it can be merged with the command above `composer update --with-dependencies spryker/kernel-app:'^1.2.0' spryker/search-http:^0.5.0 ...`

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 17]
```diff
@@ -0,0 +1,375 @@
...
+4. Build JavaScript assets for Yves using one of the following commands:
+```bash
+npm ci && npm run yves
+```
+
+```bash
+console frontend:project:install-dependencies && console frontend:yves:build
+```
+
+5. Check that required dependency is installed:
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:124]: Why not just instruct to install it?

Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:124]: @supproduction ?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 16 above)
Comment (vol4onok) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md]: I moved it

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 18]
```diff
@@ -0,0 +1,352 @@
...
+```bash
+npm list search-insights
+```
+
+6. If the previous command returned `└── (empty)`, install `search-insights` manually:
+```bash
+npm i search-insights`.
+```
```
Comment (supproduction) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:109]:
```suggestion
5. Install required `search-insights` dependency:
```bash
npm i search-insights`.
```
```

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 17 above)
Comment (supproduction) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md]: Changed

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-for-headless-frontends.md [hunk 19]
```diff
@@ -0,0 +1,45 @@
+---
+title: Algolia Personalization for headless frontends
+description: Find out how you can enable Algolia personalization in your Spryker shop based on headless approach (custom frontend or mobile application). 
+last_updated: Nov 24, 2024
+template: howto-guide-template
+---
+
+{% info_block infoBox "Info" %}
+
+Default Spryker installation supports Algolia personalization only for YVES frontend.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-for-headless-frontends.md:10]: What's a `default Spryker installation`?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-for-headless-frontends.md [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+
+{% endinfo_block %}
+
+The events from your site/application are required to enable Algolia premium features such as
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-for-headless-frontends.md:14]: Do you mean that the events need to be sent to Algolia?

Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-for-headless-frontends.md:14]: right

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-for-headless-frontends.md (hunk 19 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization-for-headless-frontends.md:10]: we can rephrase to "we have code for collecting events OOTB only for Yves", for custom headless frontends we don't have anything OOTB

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 21]
```diff
@@ -0,0 +1,355 @@
...
+
+5. Install required `search-insights` dependency:
+```bash
+npm i search-insights`.
+
+
+### Check your Yves's compatibility with new functionality
+
+In case of customizations, your codebase could have updated Yves templates on the project level (`src/Pyz/Yves/`).
+It could be a reason that some events won't be triggered or triggered with incorrect data.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:109]: What do I need to do in this case?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 22]
```diff
@@ -0,0 +1,355 @@
...
+
+#### Run the project
+1. Set `TraceableEventWidgetConfig::isDebugEnabled()` set to `true`.  
+2. Run the project locally or deploy to testing environment.
+3. Open your Shop Storefront home page.
+4. Open browser's development console and check "Preserve log" in the settings.
+
+#### Check triggered events and their payload
+
+When debug mode is enabled, you can see event logs in the console to help with inspection.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:119]: How do we know it's enabled? Do I need to enable it?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 23]
```diff
@@ -0,0 +1,355 @@
...
+  * If the home page has products, click on a product - `PRODUCT_CLICK`.
+  * If the home page has the add to cart button, click on it - `ADD_TO_CART`
+* Open a product's details page, you should see events for the actions:
+  * `PAGE_LOAD` with the product's SKU.
+  * `ADD_TO_CART` with the product's SKU, currency, price and quantity, when user clicks **Add to cart**.
+  * `ADD_TO_SHOPPING_LIST` with product SKU when user clicks **Add to shopping list**.
+  * `ADD_TO_WISHLIST` with product SKU when user clicks **Add to wishlist**.
+* Open a **Category** or **Search Results** page:
+  * `QueryID` should be present in the event payload on this page type.
+  * `PAGE_LOAD` with displayed products SKUs and displayed search filters.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:132]: What should happen to this `PAGE_LOAD`?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 24]
```diff
@@ -0,0 +1,355 @@
...
+
+When debug mode is enabled, you can see event logs in the console to help with inspection.
+
+Monitor the browser's console and execute following cases:
+* Open the home page.
+  * If the home page has products, click on a product - `PRODUCT_CLICK`.
+  * If the home page has the add to cart button, click on it - `ADD_TO_CART`
+* Open a product's details page, you should see events for the actions:
+  * `PAGE_LOAD` with the product's SKU.
+  * `ADD_TO_CART` with the product's SKU, currency, price and quantity, when user clicks **Add to cart**.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:127]: Do I need to click Add to cart to see this event?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 25]
```diff
@@ -0,0 +1,355 @@
...
+* Open Cart page
+  * (if applicable) add a new product from add to cart widget `ADD_TO_CART`
+  * (if applicable) save cart items to a shopping list `ADD_TO_SHOPPING_LIST`
+* Open Quick Order page
+  * (if applicable) add a new product from add to cart widget `ADD_TO_CART`
+  * (if applicable) save cart items to a shopping list `ADD_TO_SHOPPING_LIST`
+* Open Order Success page
+  * `PAGE_LOAD` with currency, order total, skus, prices, quantities  of purchased products.
+
+To view a full list of available events, refer to the `traceable-events-algolia` [Readme file](https://github.com/spryker-shop/traceable-event-widget/src/SprykerShop/Yves/TraceableEventWidget/Theme/default/components/molecules/traceable-events-algolia/README.md).
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:145]: 404 on the link

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 26]
```diff
@@ -0,0 +1,355 @@
...
+#### Common Issues and Solutions
+
+##### Prerequisites
+
+If you need to add, modify, or fix events at the project level, start with these two steps:
+
+- Locate the page template or view that is used for the current page.
+- Override the `{% raw %}{% block eventTracker %}{% endraw %}` block in your project’s template at the [project level](https://docs.spryker.com/docs/dg/dev/frontend-development/202410.0/yves/atomic-frontend/managing-components/overriding-components.html#create-component-folder-on-project-level).
+
+For comprehensive details about the **event configuration API**, visit the [traceable-events-orchestrator README](https://github.com/spryker-shop/traceable-event-widget/src/SprykerShop/Yves/TraceableEventWidget/Theme/default/components/molecules/traceable-events-orchestrator/README.md).
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:157]: How is this related to this section?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 27]
```diff
@@ -0,0 +1,355 @@
...
+
+    {{ parent() }}
+{% raw %}{% endblock %}{% endraw %}
+```
+
+2. Dynamic Data
+
+For adding dynamic data, refer to the [API documentation](https://github.com/spryker-shop/traceable-event-widget/src/SprykerShop/Yves/TraceableEventWidget/Theme/default/components/molecules/traceable-events-orchestrator/README.md). Adjust the configuration as needed for specific triggers.
+
+```twig
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:239]: Add file path

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 28]
```diff
@@ -0,0 +1,355 @@
...
+                  category: VALUE, // 'category' data from the '.metadata-row .product-category' element
+              },
+              ...
+          ]
+      }
+  }
+#}
+```
+
+##### Examples of the feature integration into Spryker demo shops
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:311]: What do I need to do with this?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 29]
```diff
@@ -0,0 +1,355 @@
...
+
+- [B2C Demo Shop](https://github.com/spryker-shop/b2c-demo-shop/pull/595/files)
+- [B2C Marketplace Demo Shop](https://github.com/spryker-shop/b2c-demo-marketplace/pull/474/files)
+- [B2B Demo Shop](https://github.com/spryker-shop/b2b-demo-shop/pull/542/files)
+- [B2B Marketplace Demo Shop](https://github.com/spryker-shop/b2b-demo-marketplace/pull/490/files)
+
+  
+
+### Update website agreement text
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:320]: Which section is this section related to? Updating Spryker Shop or issues?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 30]
```diff
@@ -0,0 +1,355 @@
...
+
+You should update the website agreement text and ask for user consent to have their interactions with the website tracked and sent to Algolia.
+Something similar to
+
+> **User Data analytics**
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:325]: Do you mean privacy policy?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 31]
```diff
@@ -0,0 +1,355 @@
...
+
+### Test it
+
+1. Deploy to testing environment.
+2. Make sure that Algolia is connected and configured in the Backoffice > Apps.
+
+{% info_block warningBox "Make sure" %}
+
+If you previously had ACP Algolia App connected and used, you will need to disconnect and connect it again with the same Algolia credentials in the ACP App Catalog.
+This action will update your Spryker shop config to be able to send events to Algolia.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:338]: This is confusing. Would I be able to install algolia personalization if I didn't have algolia connected and configured?  Would it even make sense to install personalization before installing algolia itself?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 32]
```diff
@@ -0,0 +1,355 @@
...
+
+{% endinfo_block %}
+
+3. Open Yves, act as a guest and logged-in user, do searches, filter results, open product pages after search, add products to cart, do order placement.
+4. Go to [Algolia Dashboard](https://dashboard.algolia.com/) and open Events from Data Sources section - `https://dashboard.algolia.com/apps/$APP_ID$/events/debugger`.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:343]: The link is pointing towards the events>debugger section. Should i go there?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 21 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:109]: This paragraph is part of "Check your Yves's compatibility with new functionality", you need to follow the steps from the next sub-sections
- Run the project
- Check triggered events and their payload

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 22 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:119]: see the previous paragraph with the title "Run the project", Set TraceableEventWidgetConfig::isDebugEnabled() set to true.

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 23 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:132]: yes

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 27 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:239]: fixed the link for "API documentation". Did you mean this?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 28 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:311]: you need to select a demoshop that your project used as starting point and check this PRs, it should help you to integrate events faster. So use it as the reference or guide

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 29 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:320]: Updating Spryker Shop

Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:320]: there is a difference between level of headers `###` (h3) or `####` (h4)

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 30 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:325]: I took it from Vasilikiy's message

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 31 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:338]: This is for customers who have already used Algolia without personalization and then decided to enable also personalization. So they need to re-connect the app after these integration guide steps.

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 32 above)
Comment (matweew) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:343]: the link has a placeholder APP_ID, so it's not a real link that can be used in the documentation, we can just remove it and write about where to find it in the Algolia 
image

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md [hunk 33]
```diff
@@ -0,0 +1,355 @@
...
+  * `PAGE_LOAD` with currency, order total, skus, prices, quantities  of purchased products.
+
+To view a full list of available events, refer to the `traceable-events-algolia` [Readme file](https://github.com/spryker-shop/traceable-event-widget/src/SprykerShop/Yves/TraceableEventWidget/Theme/default/components/molecules/traceable-events-algolia/README.md).
+
+
+#### Common Issues and Solutions
+
+##### Prerequisites
+
+If you need to add, modify, or fix events at the project level, start with these two steps:
```
Comment (andriitserkovnyi) [inline docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:152]: Would it be correct to say that most solutions in this section are related to adding, changing, or fixing events?

File: docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md (hunk 27 above)
Comment (andriitserkovnyi) [reply docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md:239]: I'm referring to this twig file. Is this any particular file or just an example?
